api/pkg/apis/v1alpha1/providers/stage/script/script_test.go

```diff
@@ -61,3 +61,15 @@ func TestShellScriptOnline(t *testing.T) {
 	assert.Nil(t, err)
 	os.Remove("staging/go1.21.6.src.tar.gz")
 }
+
+func TestShellScriptNotFoundOnline(t *testing.T) {
+	provider := ScriptStageProvider{}
+	err := provider.Init(ScriptStageProviderConfig{
+		Name:          "test",
+		Script:        "test.ps1",
+		ScriptEngine:  "powershell",
+		ScriptFolder:  "https://bing.com",
+		StagingFolder: "staging",
+	})
+	assert.NotNil(t, err)
```

Comment:
You need to check the error message since your fixes in this PR is improving the error message.

Reply:
Added type check and message check.